tests/utils.py

```diff
@@ -49,3 +51,99 @@
     '_id': 'yes! yes! yes!',
     'count': 0
 }
+
+
+TEST_SCHEMA = update_schema(BASEXMLSCHEMA, {
+    "properties": {
+        "title1": "//dc:title/node()",
+        "title2": ["//dc:title/node()", lambda x: x.lower()],
+        "title3": ["//dc:title/node()", "//dc:title/node()", lambda x, y: x + y.lower()]
```

Comment:
Then most people wouldn't have to use them, but we need to support them.
Talk in person for more.

On Tue, Mar 10, 2015 at 12:07 PM, Fabian von Feilitzsch <
notifications@github.com> wrote:

> In tests/utils.py
> <https://github.com/fabianvf/scrapi/pull/102#discussion_r26135916>:
>
> > @@ -49,3 +51,99 @@
> >      '_id': 'yes! yes! yes!',
> >      'count': 0
> >  }
> > +
> > +
> > +TEST_SCHEMA = update_schema(BASEXMLSCHEMA, {
> > +    "properties": {
> > +        "title1": "//dc:title/node()",
> > +        "title2": ["//dc:title/node()", lambda x: x.lower()],
> > +        "title3": ["//dc:title/node()", "//dc:title/node()", lambda x, y: x + y.lower()]
>
> I think the downside would be that specifying named selectors would be too
> verbose to do inline, which would complicate the schema specification. I'm
> not sure that named selectors would necessarily make the code more readable.
>
> This is what I imagine the usage looking like, and in both situations I
> think just having args is still simpler to read.
>
> def process_title(t, title=None):
>     return t.lower() + title.lower() if title else t.lower()
>
> title2_kwargs = {
>     "title": "//dc:title/node()"
> }
>
> SCHEMA = update_schema(BASEXMLSCHEMA, {
>     "properties": {
>         "title1": (("//dc:title/node(),), {"title1": "//dc:title/node()"}), process_title),
>         "title2": (("//dc:title/node(),), title2_kwargs), process_title)
>     }
> )
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/fabianvf/scrapi/pull/102/files#r26135916>.
>